shared/params/loader.go

```diff
@@ -19,6 +19,10 @@ func LoadChainConfigFile(chainConfigFileName string) {
 	// Convert 0x hex inputs to fixed bytes arrays
 	lines := strings.Split(string(yamlFile), "\n")
 	for i, line := range lines {
+		// Ignore for deposit contract address - configuration object defines it as a string.
```

Comment:
Why ignore deposit contract address from config yaml? Is the reason solely due to string?

I'm concernd users will report this as a bug

Reply:
We are **not** ignoring it from YAML config (we actually **ignored** it before, and this PR makes sure it is used, and users can pass that paramter *in config file* and override our hard-coded values). So, it is ignored from "conversion" (should I update the comment, it seems it is confusing??).

Now why we do not need to convert (ignore from conversions), it is because we use `0x` number as string, our code expects is as string:
https://github.com/prysmaticlabs/prysm/blob/ff64fdcfb53047948e558b9ca2b8fe1d41a60977/shared/params/network_config.go#L32

So, I do not need to convert, we are already in format our code expects (string).
The reason we have that string -> []byte conversion is because other params, like `GENESIS_FORK_VERSION` or `BLS_WITHDRAWAL_PREFIX` are  fixed byte arrays:
https://github.com/prysmaticlabs/prysm/blob/ff64fdcfb53047948e558b9ca2b8fe1d41a60977/shared/params/config.go#L112-L113